packages/devtools_app/lib/src/flutter/device_dialog.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:pedantic/pedantic.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../connected_app.dart';
+import '../globals.dart';
+import '../info/info_controller.dart';
+import '../table_data.dart';
+import '../utils.dart';
+import '../version.dart';
+import 'auto_dispose_mixin.dart';
+import 'common_widgets.dart';
+import 'table.dart';
+import 'theme.dart';
+
+class DeviceDialog extends StatelessWidget {
+  const DeviceDialog({
+    @required this.connectedApp,
+    @required this.flutterVersion,
+  });
+
+  final ConnectedApp connectedApp;
+  final FlutterVersion flutterVersion;
+
+  @override
+  Widget build(BuildContext context) {
+    const boldText = TextStyle(fontWeight: FontWeight.bold);
+    final theme = Theme.of(context);
+
+    final vm = serviceManager.vm;
+
```

Comment:
Can you give an example what we expect the output to be to explain the substring code

Reply:
Sure (and I'll add as a comment as well):

Here's the full string: `Dart VM version: 2.9.0-13.0.dev (dev) (Fri May 29 15:59:05 2020 +0200) on "macos_x64"`; we just want the portion before the first space.